test/react-util/global-id.spec.ts

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { expect } from 'test-drive-react';
+import { getLocalId, getRootId, separator } from '../../src/react-util/global-id';
+
+interface TestProps {
+    id?: string;
+}
+
+class TestClass extends React.Component<TestProps> {}
+
+describe('GlobalID', () => {
+   describe('getRootId', () => {
+       const id = 'pancake';
+
+        it('resolves the id from props in case it was passed', () => {
+            const testClass = new TestClass({id});
+
+            expect(getRootId(testClass)).to.equal(id);
+        });
+
+        it('resolves the id from a component\'s private state (if relevant) if it was not provided in props', () => {
+            const firstInstance = new TestClass();
```

Comment:
don't compare results to verbode values ('0', '1' etc.):

one test should check that `typeof` is string.
 
another test should check for consistency ('stickiness') of generated ID, meaning two calls on same instance should produce the same ID

another test for uniqueness,  meaning two calls on different instances should produce different IDs

Reply:
this is what i was trying to test but i agree that comparing the results of the invocations is clearer. will modify

Reply:
never, i understood now. refactoring